### PR TITLE
Separate handling of read and write contract functions

### PR DESCRIPTION
**Pull request by vishal-k25**
This pull request introduces changes by handling the function calls by differentiating between read (view/pure) and write functions.

### CODE REVIEW

File: dist/contract/index.js [hunk 1]
```diff
@@ -26,6 +26,14 @@ class VerifiedContract {
         this.abiInterface = new ethers_1.utils.Interface(abi);
         this.contract = new ethers_1.ethers.Contract(address, this.abiInterface, signer);
     }
+    isReadFunction(functionName, ...args) {
```
Comment (Mohzcrea8me) [inline dist/contract/index.js:29]: ...args can be removed here

File: src/contract/index.ts [hunk 2]
```diff
@@ -31,14 +31,25 @@ export class VerifiedContract {
...
-    private abiInterface: ContractInterface;
+    private abiInterface: utils.Interface;
 
     constructor(address: string, abi: string, signer: VerifiedWallet | Signer) {
         this.signer = signer;
         this.abiInterface = new utils.Interface(abi)
         this.contract = new ethers.Contract(address, this.abiInterface, signer);
     }
 
+    private isReadFunction(functionName: string, ...args: any): boolean {
```
Comment (Mohzcrea8me) [inline src/contract/index.ts:42]: ...args can be removed here

File: package.json [hunk 3]
```diff
@@ -34,10 +34,13 @@
     "@biconomy/account": "^4.1.1",
     "ethereumjs-tx": "^2.1.2",
     "ethers": "^5.7.2",
-    "tslib": "^2.6.2"
+    "sinon": "^18.0.0",
```
Comment (Mohzcrea8me) [inline package.json:37]: i understand sinon is needed for your testcase, we can't add new dependencies to sdk especially the ones we can do without, we need to remove sinon and your testcase now that your it runs without errors. viem has been updated as well to "viem": "^2.12.0" on my recent update

File: package.json [hunk 4]
```diff
@@ -34,10 +34,13 @@
...
+    "tslib": "^2.6.2",
+    "viem": "^2.11.1"
   },
   "devDependencies": {
```
Comment (Mohzcrea8me) [inline package.json:41]: sinon needs to be removed here too

File: src/test/functionCallsTest.spec.ts
Comment (Mohzcrea8me) [inline src/test/functionCallsTest.spec.ts:1]: nice work with the test you can remove it and edit src/test/readFunctions.spec.ts to add more read functions(doesn't have to base sepolia anymore and you can monitor the console logs to ensure read function didn't call userOp)

File: package.json [hunk 5]
```diff
@@ -34,10 +34,13 @@
...
     "@types/mocha": "^10.0.6",
+    "@types/sinon": "^17.0.3",
```
Comment (Mohzcrea8me) [inline package.json:43]: sinon needs to be removed here too

**Review by Mohzcrea8me — CHANGES_REQUESTED** [review]
Nice work, the test ran as expected too. few changes are required before merging @vishal-k25

File: package.json (hunk 3 above)
Comment (vishal-k25) [reply package.json:37]: Can I still use sinon in my tests without adding in package.json.

File: src/test/functionCallsTest.spec.ts
Comment (vishal-k25) [reply src/test/functionCallsTest.spec.ts:1]: run tests for other  read functions. no errors, it works perfectly fine without calling userOp. 
How many read functions to include in the test?

**Pull request closed without merging**